pkg/envoy/rbac/policy.go

```diff
@@ -0,0 +1,117 @@
+package rbac
+
+import (
+	xds_rbac "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	xds_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	"github.com/pkg/errors"
+)
+
+// Generate constructs an RBAC policy for the policy object on which this method is called
+func (p *Policy) Generate() (*xds_rbac.Policy, error) {
+	policy := &xds_rbac.Policy{}
+
+	// Construct the Principals ------------------------
+	var finalPrincipals []*xds_rbac.Principal
+
+	// Each RuleList follows OR semantics with other RuleList in the list of RuleList
+	for _, principalRuleList := range p.Principals {
+		// 'principalRuleList' corresponds to a single Principal in an RBAC policy.
+		// This Principal can be defined in terms of one of AND or OR rules.
+		// When AND/OR semantics are not required to define multiple rules corresponding
+		// to this principal, a single Rule in either the AndRules or OrRules will suffice.
+
+		if len(principalRuleList.AndRules) != 0 && len(principalRuleList.OrRules) != 0 {
```

Comment:
Assuming we do NOT want to have And and Or in the same RuleList object in upcoming iterations, would it not be better to disallow it by design? (by typifying RuleList or polymorph it and front it with an interface?)

Reply:
Fronting it with an interface could be confusing for clients. Could you give an example of what you have in mind?

Reply:
One alternative I can think of is 
```
type RulesList struct {
    Rules []Rules
    semantic RuleSemantic // AND or OR
}
```
We can do this in the future when we are certain a combination of AND/OR rules is not going to be possible. I am not certain of that at this point given there are not many consumers of RBAC.

Reply:
Discussed with @eduser25 offline.